src/Components/Dropdowns/Composed/CreatableTypeAheadDropdown.tsx

```diff
@@ -0,0 +1,222 @@
+// Libraries
+import React, {
+  ChangeEvent,
+  forwardRef,
+  MouseEvent,
+  useState,
+  useEffect,
+} from 'react'
+
+// Components
+import {Dropdown, DropdownRef} from '../'
+import {DropdownHeader} from '../DropdownHeader'
+import {Input} from '../../Inputs/Input'
+
+// Types
+import {
+  ComponentSize,
+  ComponentStatus,
+  DropdownMenuTheme,
+  IconFont,
+  StandardFunctionProps,
+} from '../../../Types'
+import {MenuStatus} from '../Dropdown'
+
+export interface CreatableTypeAheadDropdownProps extends StandardFunctionProps {
+  /** Text to render in input as currently selected or typed option */
+  selectedOption: string
+  /** List of options to render in menu */
+  options: string[]
+  /** Fires when an option is clicked, used to update state */
+  onSelect: (option: string) => void
+  /** Optional placeholder text when no option is selected */
+  placeholder?: string
+  /** Optional status of input */
+  inputStatus?: ComponentStatus
+  /** Optional size of input */
+  inputSize?: ComponentSize
+  /** Optional icon to be displayed to the left of text in input  */
+  inputIcon?: IconFont
+  /** Optional color preview to be displayed to the left of text.
+   * The color is determined by the selected or typed option in #000000 format.
+   * If both icon and this props are set, icon will take priority */
+  inputColorPreviewOn?: boolean
+  /** Optional theme of menu */
+  menuTheme?: DropdownMenuTheme
+  /** Optional maximum pixel height menu */
+  menuMaxHeight?: number
+  /** Optional customized dropdown item */
+  customizedDropdownItem?: (displayText: string) => JSX.Element
+}
```

Comment:
i know that this was copied over from the existing `TypeaheadDropdown` but for the future: in general, comments that restate the obvious aren't necessary and can actually make code harder to read. 

examples of comments stating the obvious:
```ts
 /** Fires when an option is clicked, used to update state */
onSelect: (option: string) => void

 /** Optional status of input */
inputStatus?: ComponentStatus

/** Optional placeholder text when no option is selected */
placeholder?: string

/** Optional theme of menu */
menuTheme?: DropdownMenuTheme
```

i think this:
```ts
  /** Optional color preview to be displayed to the left of text.
   * The color is determined by the selected or typed option in #000000 format.
   * If both icon and this props are set, icon will take priority */
```
is an example of a good comment. it provides context that may not be obvious from reading the variable name :+1: